code/segmentedtickerplant/pubsub.q

```diff
@@ -22,40 +22,40 @@ endp:{
 // Function to send end of day messages to subscribers      
 // Assumes that .u.end has been defined on the client side   
 end:{
-  (neg raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\:(`.u.endp;x;y);
+  (neg raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\:(`.u.end;x;y);
  };
 
 suball:{
-  delhandle[x].z.w;
-  :add[x];
+  delhandle[x;.z.w];
+  add[x];
+  :(x;schemas[x]);
  };
 
 subfiltered:{[x;y]
-  delhandlef[x].z.w;
-  if[11=type y;:selfiltered[x;y]];
-  if[99=type y;:addfiltered[x;y]];
+  delhandlef[x;.z.w];
+  if[11=type y;selfiltered[x;y]];
+  if[99=type y;addfiltered[x;y]];
+  :(x;schemas[x]);
  };
 
 // Add handle to subscriber in sub all mode
 add:{
-  if[not (count subrequestall x)>i:subrequestall[x;]?.z.w;
+  if[not (count subrequestall x)>i:subrequestall[x]?.z.w;
     subrequestall[x],:.z.w];
-  (x;$[99=type v:value x;v;0#v])
  };
 
 // Add handle to subscriber in sub filtered mode
+// Where clause and column filters are parsed before adding to subrequestfiltered table
 addfiltered:{[x;y]
-  if[not .z.w in subrequestfiltered[x]`handle;
-    @[`.stpps.subrequestfiltered;x;:;(enlist[`handle]!enlist .z.w),y[x]]];
-  (x;$[99=type v:value x;v;0#v])
+  filts:$[null y[x]`filts;();enlist parse string y[x]`filts];
```

Comment:
Could we make the parsing optional on type? So parse it if it's a string, if it's a mixed list just treat it as a functional form where clause (at the user's risk). Some q based subscribers might prefer that approach. Also, need to error trap the parse to protect against nonsense being inserted.